### PR TITLE
Java translated version using JNA

### PR DESCRIPTION
**Pull request by agaertner**
I translated the listener.cp to java using the JNA library. Add it to your (this) repository, because creating a new one is horrible and not convenient.

### CODE REVIEW

File: MSNeventListener.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        {
+            if (uMsg == WM_COPYDATA) {
+
+                COPYDATASTRUCT copydatastruct = new COPYDATASTRUCT(new Pointer(lParam.longValue()));
+
+                /* Split message data at zero bytes. */
+                String[] arr = copydatastruct.lpData.getWideString(0).split("\\\\0");
+                
+                /* End the process if the message data is a duplicate of the previous. */ 
+                if (Arrays.deepEquals(arr, _msnArray)) { return new LRESULT(0); };
```
Comment (Piotrekol) [inline MSNeventListener.java:179]: Duplicate handling shouldn't be there and on top of that it's handled incorrectly. 
Assume player retrys already played map - request gets ignored.

Comment (Piotrekol) [reply MSNeventListener.java:179]: If you still want to include that detection there, do not ignore retrys/changes to map with the same exact name.
This is how I handled it in c# https://github.com/Piotrekol/StreamCompanion/blob/master/osu!StreamCompanion/Code/Core/Maps/Processing/MapStringFormatter.cs#L78-L110

**Comment by agaertner** [comment]
You are wrong. Since a retry will swap beweeten Arrays containing the word
Listening and playing. A retry will be detected.  I tested it.  Also the
second handling can be savely ignored,  but since I needed it this way for
phantombot to work. Check phantombot forums custom scripts section for
osu!NowPlaying if you wanna try it.

Am 08.02.2017 4:06 nachm. schrieb "Piotrekol" <notifications@github.com>:

**Comment by agaertner** [comment]
The dublicate check just ignores the times osu sends a msg containing
listening. That is happening when you are in main menu and again when you
swap to mao select. Your osu companion doesnt need to check for that,  but
since the phantombot replies directly to chat its an anti spam measure.

Am 08.02.2017 4:06 nachm. schrieb "Piotrekol" <notifications@github.com>:

**Pull request closed without merging**